src/session-client.ts

```diff
@@ -101,4 +127,24 @@ class SRPClientSessionStep2 {
       throw new Error("Bad server credentials");
     }
   }
+
+  public toJSON(): SRPClientSessionStep2State {
+    return {
+      A: this.A.toString(16),
+      M1: this.M1.toString(16),
+      S: this.S.toString(16),
+    };
+  }
+
+  public static fromState(
+    routines: SRPRoutines,
+    state: SRPClientSessionStep2State,
+  ) {
+    return new SRPClientSessionStep2(
+      routines,
+      BigInt("0x" + state.A),
```

Comment:
Adding 0x is kinda annoying. We could just use base 10, but I thought hex makes more sense for bigint. What do you think?

Reply:
either way is fine